### ISSUE TITLE
Cap neutralization problem

### ISSUE THREAD

**Issue by SubjectXXX** [pyfa-org/Pyfa#1507]
## Bug Report
Maybe this just a question that needs answering, but it seems that scenario, where one ship gets neuted by another is a bit vague.

In pyfa, cap management is represented with two values: cap/s for recharge and -cap/s for usage.
Neutralizers (and NOS) drain capacitor at the end (or start) of their cycle.

Is the calculation done on this important fact or does projection of a ship fitting that uses neutralizers/NOS only deduct -cap/s value from neutralizer module from cap/s recharge rate of tested ship fit?

In other words: I think, there is a difference (coz of not linear natural recharge rate of capacitor), if you simply deduct cap/s neutralized from cap/s recharge rate of a ship in comparison to deducting whole value of neutralized capacitor at the end (start) of a module cycle and simulate how much capacitor will recharge while neutralization module is cycling for another cap deduction.

Which method is pyfa using?

Same question stands for using Capacitor booster module: is cap injected every second or is it injected at once and then cap recharge kicks in while module cycles??

I know its hard to simulate different start cycles of modules that use capacitor and hostile modules that neutralize capacitor, but I'm sure that we could simulate worst case scenario of cap usage with deducting all the cap usage from capacitor pool at the same time (triggering modules all at the same time, having in mind different cycle usage for neuts and NOS for example) thus trying to put capacitor below its peak recharge rate..

If this is wrong place to raise this kind of question, please delete this ticket and point me to the correct place. I have this text saved :)

**Comment by SubjectXXX** [comment]
Found similar topic for investigation:
https://github.com/pyfa-org/Pyfa/issues/1401
But I dont think it points to the same question at the end..

**Comment by blitzmann** [comment]
@Neugeniko would you be able to take a peak at this since you were more recently in this code? :)

**Comment by Neugeniko** [comment]
We simulate all modules (local and projected) as using/reducing capacitor at the start of their cycle. We stagger the start of similar modules so there is cap regeneration between their activation too. And cap recharge is calculated continuously between activation of modules for a full day.

**Comment by SubjectXXX** [comment]
Yes, I think that answered my question just fine. 

Maybe just on a detail that simulation uses start of module cycle to determine cap deduction: neut and nos have different "drainage" times. So if I project a fit with neut and nos, neut should take away cap at time 0, while nos should drain cap at time 0+ n (n being cycle length) (or does nos start at the start of a cycle and neut at the end, not sure..). Between time 0 and time 0+n, there is a natural recharge rate going on or cap injection (when does cap injection happen? After cap deduction from modules or neuts/noses in simulation?). If simulation deducts both cap drainage values at the same time, at the start of a cycle, one can quickly drop below 33% peak recharge.

Numbers could be a bit different in this case, not sure how hard it is to implement it tho..

**Comment by Neugeniko** [comment]
Also i'm pretty sure the neutralizers and nos does all the cap adjustment at the start of cycle.

From euni -

When activating an Energy Neutralizer on a target, the activation cost will immediately be deducted from the neutralizer's capacitor and the neutralization amount (modified, if in falloff range) will be immediately deducted from the target's capacitor. This will be repeated each cycle time for as long as the neut is activated.

**Comment by SubjectXXX** [comment]
No, they are different for sure, just not sure which one at the start and which one at the end..

**Comment by Neugeniko** [comment]
I would have to test to find out for sure.

**Comment by Neugeniko** [comment]
Armor boost happen at the end of cycle and shield at the start, you sure thats not what your thing about?

**Comment by SubjectXXX** [comment]
I tested it before latest patch. I'm at work atm, so cant retest it. If you dont have answer to it, I'll test it in the evening and let you know.

**Comment by Neugeniko** [comment]
That would be great. Looking forward to your report.

**Comment by SubjectXXX** [comment]
Hehe.. Was just writing the same.
